### ISSUE TITLE
Getting error (after upgrade to 5.13.29?)

### ISSUE THREAD

**Issue by samteezy** [misenhower/homebridge-unifi-led-control#1]
In the last couple days this has stopped working for me.

The error I'm getting is: 
`Error: write EPROTO 140069852904776:error:1409442E:SSL routines:ssl3_read_bytes:tlsv1 alert protocol version:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 70`

I'm wondering if this has anything to do with [the last item in the Ubiquiti release notes regarding the regeneration of self-signed certificates](https://community.ui.com/releases/UniFi-Network-Controller-5-13-29/d7647910-77a2-4e61-bbfe-389206f2d6ad). When I navigate to the controller in my browser, the certificate is TLS 1.0.

Full log below:
```
[6/13/2020, 12:10:37 PM] [UniFi] Error, logging in and trying again
[6/13/2020, 12:10:37 PM] [UniFi] Error: write EPROTO 140069852904776:error:1409442E:SSL routines:ssl3_read_bytes:tlsv1 alert protocol version:../deps/openssl/openssl/ssl/record/rec_layer_s3.c:1544:SSL alert number 70

    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:92:16) {
  errno: 'EPROTO',
  code: 'EPROTO',
  syscall: 'write',
  config: {
    url: 'api/login',
    method: 'post',
    data: '{"username":"homebridge","password":"L2&Lq%2m%yScgr"}',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json;charset=utf-8',
      cookie: '',
      'User-Agent': 'axios/0.19.2',
      'Content-Length': 53
    },
    baseURL: 'https://10.10.1.21:8443',
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    adapter: [Function: httpAdapter],
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    validateStatus: [Function: validateStatus],
    httpsAgent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 443,
      protocol: 'https:',
      options: [Object],
      requests: {},
      sockets: [Object],
      freeSockets: {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      maxCachedSessions: 100,
      _sessionCache: [Object],
      [Symbol(kCapture)]: false
    }
  },
  request: Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      bufferedRequest: null,
      lastBufferedRequest: null,
      pendingcb: 0,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: false,
      bufferedRequestCount: 0,
      corkedRequestsFree: [Object]
    },
    writable: true,
    _events: [Object: null prototype] {
      response: [Function: handleResponse],
      error: [Function: handleRequestError]
    },
    _eventsCount: 2,
    _maxListeners: undefined,
    _options: {
      protocol: 'https:',
      maxRedirects: 21,
      maxBodyLength: 10485760,
      path: '/api/login',
      method: 'POST',
      headers: [Object],
      agent: [Agent],
      agents: [Object],
      auth: undefined,
      hostname: '10.10.1.21',
      port: '8443',
      nativeProtocols: [Object],
      pathname: '/api/login'
    },
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 53,
    _requestBodyBuffers: [ [Object] ],
    _onNativeResponse: [Function],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 6,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: null,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      connection: [TLSSocket],
      _header: 'POST /api/login HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/json;charset=utf-8\r\n' +
        'cookie: \r\n' +
        'User-Agent: axios/0.19.2\r\n' +
        'Content-Length: 53\r\n' +
        'Host: 10.10.1.21:8443\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      insecureHTTPParser: undefined,
      path: '/api/login',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      _redirectable: [Circular],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    _currentUrl: 'https://10.10.1.21:8443/api/login',
    [Symbol(kCapture)]: false
  },
  response: undefined,
  isAxiosError: true,
  toJSON: [Function]
}
(node:4064) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'data' of undefined
    at UniFiDevice.setProperties (/homebridge/node_modules/homebridge-unifi-led-control/src/UniFiDevice.js:121:40)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:4064) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 43)
```

**Comment by misenhower** [comment]
I haven't run into this myself on controller version 5.14.9, but I think you're on the right path with the TLS version. Mine is using TLS 1.2 and it looks like nodejs currently sets 1.2 as the minimum version.

I'm wondering if you have some outdated settings in your controller's config. I found [this old thread](https://community.ui.com/questions/UniFI-Controller-TLS-1-2/a1a42053-de1f-4cce-bf2b-187ed672536e) that might help bump up the TLS version on your controller.

Otherwise I might be able to tell NodeJS to accept an older TLS version but it would probably be best to see if you can get your controller up-to-date first. Let me know 😃

**Comment by samteezy** [comment]
Yep, I had found the same thread as you and this morning implemented the tweaks to update to TLS 1.2. After restarting the controller, Homebridge is successfully talking with the APs again. Thanks!
